app/controllers/tags_controller.rb

```diff
@@ -34,41 +38,92 @@ def create
     @tag = Tag.new(tag_params)
     authorize @tag
 
-    respond_to do |format|
-      if @tag.save
-        format.html { redirect_to @tag, notice: 'Tag was successfully created.' }
-        format.json { render :show, status: :created, location: @tag }
-      else
-        format.html { render :new }
-        format.json { render json: @tag.errors, status: :unprocessable_entity }
+    @tag.transaction do
+      begin
+        @tag.save!
+        if(@tag.category == "position")
```

Comment:
@Vexxenn For the create, you can use the function that already exists in the model for consistency `Tag.create_start_end_position_tags`

Reply:
@elubow We can keep the method i will just change `save` function to `save!` since that way it can trigger a RecordInvalid error in order to start a rollback if something goes wrong.

Reply:
@Vexxenn Thanks. I never remember the differences between the `!` and non-`!` methods. Good catch on this.